pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -323,6 +327,29 @@ func applyTrafficPolicy(cluster *v2.Cluster, policy *networking.TrafficPolicy, p
 	applyUpstreamTLSSettings(cluster, tls)
 }
 
+func applyCommonHttpProtocolOptions(cluster *v2.Cluster) {
+	idletimeout := intEnv("IDLE_TIMEOUT", 0)
```

Comment:
This should go as a global mesh config option in api/mesh/config.proto. You should then read it here and apply to the config. 
See how connect timeout is implemented.